### PR TITLE
Use weak divergence in scalar advection

### PR DESCRIPTION
**Pull request by simonbyrne**
## Purpose 
If we use weak divergence for advection, then no mass is transported between elements until the DSS operation is performed. This means that we can guarantee that the total mass in the element is non-negative (and so don't need to borrow horizontally).

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by szy21** [comment]
bors try

File: src/prognostic_equations/advection.jl [hunk 1]
```diff
@@ -59,15 +59,23 @@ function horizontal_tracer_advection_tendency!(Yₜ, Y, p, t)
...
     end
 
+    horizontal_tracer_advection_tendency_turbconv!(
+        Yₜ,
+        Y,
+        p,
+        t,
+        p.atmos.turbconv_model,
+    )
+
```
Comment (simonbyrne) [inline src/prognostic_equations/advection.jl:72]:
```suggestion
```

**Pull request merged**